core/composables/useShapeSelect.js

```diff
@@ -139,8 +142,14 @@ const pdfConverter = {
         const scale = this.getScale(chunk);
 
         if (scale) {
-          const vertices = getNormalizedVertices(currentShape.vertices, scale);
-          shapes.push({ ...currentShape, vertices });
+          const { vertices, ...rest } = currentShape;
+          const normalizedVertices = getNormalizedVertices(vertices, scale);
+          const columnData = haveColumnsData && getColumnsData(text, columnPropertyLine, columnDataStr);
+          shapes.push({ 
+            isSelected: true,
+            vertices: normalizedVertices,
+            ...(columnData ? { data: { ...columnData, ...rest }} : { data: rest })
```

Comment:
Maybe this could be tidied up a bit?
```suggestion
            data: columnData ? { ...columnData, ...rest } : rest
```